### PR TITLE
Enforce filename casing with tslint rule

### PR DESCRIPTION
**Pull request by andrewgordstewart**
This does not enforce a directory name convention, but if I try to
create a new file `/foo/bar/FooBar/fooBar.ts` and get a lint error,
hopefully I notice that the `FooBar` directory name is also not
conventional

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,122 +5,149 @@ Magmo Apps
...
+
 #### Prettier
+
 Prettier is configured via `.prettierrc`.
 Tests will fail if code does not satisfy the rules specificied in `.prettierrc`.
 We suggest that you configure your editor to auto-format using prettier,
 or that you run it in a pre-commit git hook.
 You can also run `yarn prettier:write`.
 
+#### Naming
```
Comment (kerzhner) [inline README.md]: Per Slack conversation, Pascal case is no more.

File: packages/wallet/src/components/transaction-failed.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import SidebarLayout from './SidebarLayout';
+import SidebarLayout from './sidebar-layout';
 import { Button } from 'reactstrap';
 import magmoFireBall from '../images/white_fireball.svg';
```
Comment (kerzhner) [inline packages/wallet/src/components/transaction-failed.tsx]: Not a big deal, but looks like there are more files that do not conform to our naming convention.

**Comment by andrewgordstewart** [comment]
For those who want to understand the failing builds. (You hopefully won't have to care, with the tslint rule added in this PR, but someone might create a `src/FooBar` directory and run into this issue again.


The filesystem on OSX is, by default, case insensitive. You are free to use capital letters in your files, but the filesystem does not distinguish between an `A` and an `a` in a filename. Thus, you cannot have both `index.tsx` and `Index.tsx` in the same directory.

Because of this, `git` does not notice a change when you rename `index.tsx` to `Index.tsx` -- from the OS' point of view, the file has not changed.
```
~/Code/magmo/apps/packages/wallet on filename-convention
$ ls index.ts
index.ts

~/Code/magmo/apps/packages/wallet on filename-convention
$ g s
On branch filename-convention
Your branch is up to date with 'origin/filename-convention'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        ../../.vscode/
        .vscode/

nothing added to commit but untracked files present (use "git add" to track)

~/Code/magmo/apps/packages/wallet on filename-convention
$ mv index.ts Index.ts
index.ts -> Index.ts

~/Code/magmo/apps/packages/wallet on filename-convention
$ ls Index.ts
Index.ts

~/Code/magmo/apps/packages/wallet on filename-convention
$ git s
On branch filename-convention
Your branch is up to date with 'origin/filename-convention'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        ../../.vscode/
        .vscode/

nothing added to commit but untracked files present (use "git add" to track)
```

`git` has a `mv` command that renames a file and stages that change. It can be used to stage a case-only filename change.
```
~/Code/magmo/apps/packages/wallet on filename-convention
$ ls index.ts
index.ts

~/Code/magmo/apps/packages/wallet on filename-convention
$ g s
On branch filename-convention
Your branch is up to date with 'origin/filename-convention'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        ../../.vscode/
        .vscode/

nothing added to commit but untracked files present (use "git add" to track)

~/Code/magmo/apps/packages/wallet on filename-convention
$ git mv index.ts Index.ts

~/Code/magmo/apps/packages/wallet on filename-convention
$ ls Index.ts
Index.ts

~/Code/magmo/apps/packages/wallet on filename-convention
$ g s
On branch filename-convention
Your branch is up to date with 'origin/filename-convention'.

Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        renamed:    index.ts -> Index.ts

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        ../../.vscode/
        .vscode/
```

This allows you to commit case-only name changes to files, on an OS with a case-insensitive filesystem.

**Comment by kerzhner** [comment]
@andrewgordstewart, thanks for the explanation I enjoyed this read.

**Pull request merged**